packages/gestalt/src/SheetConfirmationPopover.js

```diff
@@ -0,0 +1,161 @@
+// @flow strict
+
+import { type Node, useEffect, useRef } from 'react';
+import { useAnimation } from './AnimationContext.js';
+import Box from './Box.js';
+import Flex from './Flex.js';
+import Popover from './Popover.js';
+import Button from './Button.js';
+import Text from './Text.js';
+import TrapFocusBehavior from './behaviors/TrapFocusBehavior.js';
+import { useDefaultLabelContext } from './contexts/DefaultLabelProvider.js';
+import { ESCAPE } from './keyCodes.js';
+
+type Props = {|
+  onDismiss: () => void,
+  anchor: ?HTMLElement,
+  dismissConfirmation?:
+    | boolean
```

Comment:
Here I don't think we need the polymorphism. When this component is rendered, we know the parent Sheet's `dismissConfirmation` is either `true` or an object. There also isn't need for those fields to be nested here. So in InternalSheet, we can do something like
```js
{showPopover && (
  <SheetConfirmationPopover
    anchor={dismissButtonRef.current}
    onDismiss={() => {
      setShowPopover(false);
      dismissButtonRef?.current?.focus();
    }}
    {...(typeof dismissConfirmation === 'object' ? dismissConfirmation : {})}
  />
)}
```
allowing for a flatter props structure here:
```js
type Props = {|
  anchor: ?HTMLElement,
  onDismiss: () => void,
  message?: string,
  subtext?: string,
  primaryAction?: {|
    accessibilityLabel?: string,
    text?: string,
    onClick?: ({|
      event:
        | SyntheticMouseEvent<HTMLButtonElement>
        | SyntheticMouseEvent<HTMLAnchorElement>
        | SyntheticKeyboardEvent<HTMLAnchorElement>
        | SyntheticKeyboardEvent<HTMLButtonElement>,
    |}) => void,
  |},
  secondaryAction?: {|
    accessibilityLabel?: string,
    text?: string,
    onClick?: ({|
      event:
        | SyntheticMouseEvent<HTMLButtonElement>
        | SyntheticMouseEvent<HTMLAnchorElement>
        | SyntheticKeyboardEvent<HTMLAnchorElement>
        | SyntheticKeyboardEvent<HTMLButtonElement>,
    |}) => void,
  |},
|};
```

Reply:
DONE